src/main/java/com/arpnetworking/tsdcore/sinks/TcpSinkActor.java

```diff
@@ -0,0 +1,282 @@
+                final Tcp.Write write = (Tcp.Write) failed.cmd();
+                final Ack ack = (Ack) write.ack();
+                // Put the message back on the front of the queue and signal that we
+                // want to start writing again
+                _pendingRequests.offerFirst(ack._data);
+
+                // Potential race where a new connection could be created
+                // before the error of a previous client was surfaced
+                if (sender().equals(_client)) {
+                    _client.tell(TcpMessage.resumeWriting(), self());
+                }
+            }
+        } else if (message instanceof Tcp.WritingResumed) {
+            _waitingForAck = false;
+            dispatchPending();
+        } else if (message instanceof Tcp.ConnectionClosed) {
+            _client = null;
+            connect();
+        } else if (message instanceof Connect) {
+            connect();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void postStop() throws Exception {
+        super.postStop();
+        LOGGER.info()
+                .setMessage("Shutdown sink actor")
+                .addData("sink", _sink)
+                .addData("recordsWritten", _recordsWritten)
+                .log();
+    }
+
+    private void connect() {
+        final TcpExt tcp = Tcp.get(context().system());
+        tcp.manager().tell(TcpMessage.connect(new InetSocketAddress(_serverAddress, _serverPort)), self());
+    }
+
+    private void processEmitAggregation(final EmitAggregation emitMessage) {
+        final PeriodicData periodicData = emitMessage.getData();
+
+        LOGGER.debug()
+                .setMessage("Writing aggregated data")
+                .addData("sink", _sink)
+                .addData("dataSize", periodicData.getData().size())
+                .addData("conditionsSize", periodicData.getConditions().size())
+                .addContext("actor", self())
+                .log();
+
+        if (!periodicData.getData().isEmpty() || !periodicData.getConditions().isEmpty()) {
+            final ByteString data = _sink.serializeData(periodicData);
+
+            // TODO(vkoskela): Add logging to client [MAI-89]
+            // TODO(vkoskela): Add instrumentation to client [MAI-90]
+
+            if (_pendingRequests.size() >= _maximumQueueSize) {
+                EVICTED_LOGGER.warn()
+                        .setMessage("Evicted data from HTTP sink queue")
+                        .addData("sink", _sink)
+                        .addData("count", 1)
+                        .addContext("actor", self())
+                        .log();
+                _pendingRequests.poll();
+            }
+
+            _pendingRequests.offer(data);
+            dispatchPending();
+        }
+    }
+
+    private void dispatchPending() {
+        if (!_waitingForAck && !_pendingRequests.isEmpty() && _client != null) {
+            // Push up to 10 messages
```

Comment:
This isn't a batch size, per se.  This is a set of bytes we're sending to Akka before we expect to get an Ok/Fail response from the TCP actor.  We could make this configurable, but I'm not sure if I see the need.